### PR TITLE
Fix dependencies

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
* Upgrading`jetty-server` (which brings in `jetty-http`) due to GHSA-cj7v-27pg-wf7q which is brought in by `hadoop-minicluster`.
* Upgrading `kotlin-stdlib` which is brought in through `hadoop-minicluster`. See GHSA-cqj8-47ch-rvvq.
* Upgrading `snakeyaml` which is brought in by `com.avast.gradle:gradle-docker-compose-plugin`. See GHSA-98wm-3w3q-mw94

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -116,6 +116,7 @@ dependencies {
   api 'org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10'
   api 'de.thetaphi:forbiddenapis:3.3'
   api 'com.avast.gradle:gradle-docker-compose-plugin:0.15.2'
+  api 'org.yaml:snakeyaml:1.32'
```
Comment (reta) [inline buildSrc/build.gradle:119]: Please `${versions.snakeyaml}`, from https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/version.properties

Comment (VachaShah) [reply buildSrc/build.gradle]: Oh yes! Updated to use from the version.properties file

**Comment by VachaShah** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/5862/
> * **CommitID:** [340f9ed](https://github.com/opensearch-project/OpenSearch/commit/340f9ed4ebf9bd5ce8b8cdae6583d9957c013dea)
>   Please examine the workflow log, locate, and copy-paste the failure below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

The log says no test failures

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -116,6 +116,7 @@ dependencies {
   api 'org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10'
```
Comment (reta) [inline buildSrc/build.gradle:116]: Sorry for being peeky, it seems like we have a number of `'org.jetbrains.kotlin` deps (like here and [in the](https://github.com/opensearch-project/OpenSearch/pull/4963/files#diff-6ffeac346fb766ef182207bf47e9f4472bae38bc46196eb14bb690de71c4e4cdR54) , could we put the version in `versions.properties` instead?

Comment (VachaShah) [reply buildSrc/build.gradle]: Makes sense! Done.

**Review by reta — APPROVED** [review]

**Comment by VachaShah** [comment]
Windows precommit failing with:

```
JVM crash log found: file:///D:/a/OpenSearch/OpenSearch/hs_err_pid1884.log

FAILURE: Build failed with an exception.

* What went wrong:
Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)
```

**Comment by VachaShah** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:**  ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/5897/
> * **CommitID:** [d7a5508](https://github.com/opensearch-project/OpenSearch/commit/d7a5508cc8f0c9d1472942bb7aa3553344f99a9d)
>   Please examine the workflow log, locate, and copy-paste the failure below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

Test failing:

```
org.opensearch.cluster.allocation.AwarenessAllocationIT.testThreeZoneOneReplicaWithForceZoneValueAndLoadAwareness
```

Flakey test reported in #3603

**Pull request merged**